### PR TITLE
feat: address input form dialog component

### PR DESCRIPTION
**Pull request by kibo-sahilarora**

### CODE REVIEW

File: components/dialogs/AddressInputFormDialog/AddressInputFormDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+import { Theme } from '@mui/material/styles'
+import { useTranslation } from 'next-i18next'
+
+import AddressForm from '@/components/common/AddressForm/AddressForm'
+import KiboDialog from '@/components/common/KiboDialog/KiboDialog'
+import { useModalContext } from '@/context/ModalContext'
+import type { Address, ContactForm } from '@/lib/types'
+
+interface AddressInputFormDialogProps {
+  formTitle?: string
```
Comment (kibo-chandradeeptalaha) [inline components/dialogs/AddressInputFormDialog/AddressInputFormDialog.tsx:13]: Try to keep the mandatory props grouped together at the top, then nullable properties

**Review by kibo-chandradeeptalaha — COMMENTED** [review]
Looks good, just few points. Please take a look

File: components/dialogs/AddressInputFormDialog/AddressInputFormDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+  isUserLoggedIn: boolean
+  setAutoFocus: boolean
+  validateForm: boolean
+  isAddressFormInDialog?: boolean
+  onSaveAddress: (data: Address) => void
+  onFormStatusChange: (status: boolean) => void
+  onHandleAddressValidationAndSave: () => void
+}
+
+const buttonStyle = {
```
[the remark this answers is not in the document]
Comment (kibo-sahilarora) [reply components/dialogs/AddressInputFormDialog/AddressInputFormDialog.tsx:25]: It validates the form on clicking 'Save' button and based on the validity status request is sent.

File: components/dialogs/AddressInputFormDialog/AddressInputFormDialog.tsx (hunk 1 above)
Comment (kibo-sahilarora) [reply components/dialogs/AddressInputFormDialog/AddressInputFormDialog.tsx]: Done.

File: components/dialogs/AddressInputFormDialog/AddressInputFormDialog.tsx [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+              {t('cancel')}
+            </Button>
+            <Button
+              variant="contained"
+              color="inherit"
+              sx={{ ...buttonStyle }}
+              {...(!isAddressFormValid && { disabled: true })}
+              onClick={onHandleAddressValidationAndSave}
+            >
+              {t('save')}
```
Comment (TriptiVermaXP) [inline components/dialogs/AddressInputFormDialog/AddressInputFormDialog.tsx:78]: 1. Save button color should be primary (green color).
2. Add '*' to country field.
3. Add new address heading should be in Capitalization.

**Review by TriptiVermaXP — APPROVED** [review]
overall looks good. Please check minor comments.

File: components/dialogs/AddressInputFormDialog/AddressInputFormDialog.tsx [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+import type { Address, ContactForm } from '@/lib/types'
+
+interface AddressInputFormDialogProps {
+  isAddressFormValid: boolean
+  isUserLoggedIn: boolean
+  setAutoFocus: boolean
+  validateForm: boolean
+  formTitle?: string
+  contact?: ContactForm
+  isAddressFormInDialog?: boolean
```
Comment (kibo-sushant) [inline components/dialogs/AddressInputFormDialog/AddressInputFormDialog.tsx:19]: As This component always going to display address form in dialog no need of this prop

File: components/dialogs/AddressInputFormDialog/AddressInputFormDialog.tsx [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+import KiboDialog from '@/components/common/KiboDialog/KiboDialog'
+import { useModalContext } from '@/context'
+import type { Address, ContactForm } from '@/lib/types'
+
+interface AddressInputFormDialogProps {
+  isAddressFormValid: boolean
+  isUserLoggedIn: boolean
+  setAutoFocus: boolean
+  validateForm: boolean
+  formTitle?: string
```
Comment (kibo-sushant) [inline components/dialogs/AddressInputFormDialog/AddressInputFormDialog.tsx:17]: As we are going to pull it from lang file, no need of this property

File: components/dialogs/AddressInputFormDialog/AddressInputFormDialog.tsx [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+  contact?: ContactForm
+  isAddressFormInDialog?: boolean
+  onSaveAddress: (data: Address) => void
+  onFormStatusChange: (status: boolean) => void
+  onHandleAddressValidationAndSave: () => void
```
Comment (kibo-sushant) [inline components/dialogs/AddressInputFormDialog/AddressInputFormDialog.tsx:22]: No need of Handle keyword here, delete it

File: components/dialogs/AddressInputFormDialog/AddressInputFormDialog.tsx [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+            </Button>
+          </Stack>
+        </>
+      }
+      customMaxWidth="35rem"
+      onClose={closeModal}
+    />
+  )
+}
+export default AddressInputFormDialog
```
Comment (kibo-sushant) [inline components/dialogs/AddressInputFormDialog/AddressInputFormDialog.tsx:88]: Missing Integration Testcases are missing (without mocking AddressForm)

File: components/dialogs/AddressInputFormDialog/AddressInputFormDialog.tsx (hunk 3 above)
Comment (kibo-sahilarora) [reply components/dialogs/AddressInputFormDialog/AddressInputFormDialog.tsx]: Done

File: components/dialogs/AddressInputFormDialog/AddressInputFormDialog.tsx (hunk 4 above)
Comment (kibo-sahilarora) [reply components/dialogs/AddressInputFormDialog/AddressInputFormDialog.tsx]: Done

File: components/dialogs/AddressInputFormDialog/AddressInputFormDialog.tsx (hunk 5 above)
Comment (kibo-sahilarora) [reply components/dialogs/AddressInputFormDialog/AddressInputFormDialog.tsx]: This prop will be used for changing the heading for 'edit address' and 'add new address'.

**Review by kibo-kevinwatts — APPROVED** [review]

**Pull request merged**